spring-cloud-data-core/src/main/java/org/springframework/cloud/data/core/ModuleDefinition.java

```diff
@@ -0,0 +1,398 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.core;
+
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.core.style.ToStringCreator;
+import org.springframework.util.Assert;
+
+/**
+ * Representation of a module in the context of a defined stream.
+ * This includes module options provided in a stream definition.
+ * This does not include module information required at deployment
+ * time (such as the number of module instances).
+ *
+ * @author Mark Fisher
+ * @author Gary Russell
+ * @author Luke Taylor
+ * @author Ilayaperumal Gopinathan
+ * @author Patrick Peralta
+ */
+public class ModuleDefinition implements Comparable<ModuleDefinition> {
+
+	/**
+	 * Name of module.
+	 */
+	private final String name;
+
+	/**
+	 * Symbolic name of a module. This may be generated to a default
+	 * value or specified in the DSL string.
+	 */
+	private final String moduleLabel;
```

Comment:
this field can be named just "label" as well to match the new-and-improved getter